packages/expo-file-system/ios/Next/FileSystemFile.swift

```diff
@@ -2,9 +2,24 @@ import Foundation
 import ExpoModulesCore
 
 internal final class FileSystemFile: FileSystemPath {
+  override init(url: URL) {
+    if !url.hasDirectoryPath {
+      super.init(url: url)
+      return
+    }
+    super.init(url: url.deletingLastPathComponent().appendingPathComponent(url.lastPathComponent, isDirectory: false))
+  }
   func create() {
     FileManager.default.createFile(atPath: url.path, contents: nil)
   }
+  func exists() -> Bool {
+    var isDirectory: ObjCBool = false
+    if FileManager.default.fileExists(atPath: url.path, isDirectory: &isDirectory) {
+      return !isDirectory.boolValue
```

Comment:
To be honest I'm not sure if this is a good idea to return `false` if it exists but as a directory 🤔 That's why I think there should be only one class for files and directories (that has `isDirectory` property).

Reply:
@byCedric @brentvatne @alanjhughes @lukmccall @tsapeta would love to hear your thoughts on this:

We have 2 classes:
- File
- Directory

They do not check if a file/directory exists on creation, so it's possible to create a Directory with a path under which a file already exists.

```
new File("documents/example").create()
const directory = new Directory("documents/example")
```

We're thinking that both files and directories should have an `.exists` method that returns a boolean.

Now, if you call `.exists` on `directory`, should it return `true` or `false`? What about a `File` class where a directory already exists at that path? I know directories are files, but none of the file methods will really work for directories.

I'm personally leaning towards returning `exists: false` and erroring out during creation, rather than `exists: true` and erroring out on each method, both for `File` where directories exist and for `Directory` where files exist.

What's you opinion on this?

Reply:
Random thoughts:

- If we ask users to create a new `File` class, I think we need to be careful not mixing it up with the Web spec's `File` class ([docs](https://developer.mozilla.org/en-US/docs/Web/API/File)).
- Looking at Node's [`fs.existsSync`](https://nodejs.org/api/fs.html#fsexistssyncpath) method, it returns `true` for directories. So if that exists, I think it might be better to return `true`, as that's pretty much the same behavior on Node.
  - That being said, Node `fs.exists` API has been deprecated, in favor of more verbose APIs ([fs.stat()](https://nodejs.org/api/fs.html#fsstatpath-options-callback) or [fs.access()](https://nodejs.org/api/fs.html#fsaccesspath-mode-callback)). We could do something similar with `File.exists` or `Directory.exists`, using more verbose APIs.
- If a directory exists at the path of a `File`, and you call `Directory.create`, it should likely error. (and the same for the `File.create` while having a directory)

Reply:
`If we ask users to create a new File class, I think we need to be careful not mixing it up with the Web spec's File class ([docs](https://developer.mozilla.org/en-US/docs/Web/API/File)).` – I think we're pretty clear on this being an import from `expo-file-system`, but I am open to any ideas on how to differentiate it more.

`If a directory exists at the path of a File, and you call Directory.create, it should likely error. (and the same for the File.create while having a directory)` – agreed, let'll throw an error in such case.

Reply:
`Looking at Node's [fs.existsSync](https://nodejs.org/api/fs.html#fsexistssyncpath) method, it returns true for directories. So if that exists, I think it might be better to return true, as that's pretty much the same behavior on Node.` – I think that's because you use the same class for files and directories, so it makes more sense there.